tests/dotnet-structuring.tests/libraryTest.cs

```diff
@@ -0,0 +1,204 @@
+using dotnet_structuring.library;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+
+namespace dotnet_structuring.tests
+{
+    public class TestSetup
+    {
+        //Setup Variables
+        Templates __ = new Templates();
+        TempDirectory temp = new TempDirectory();
+        public string CurrentLog { get; set; }
+        public string Options { get; private set; }
+        public string NETCommand { get; private set; }
+        public string ProjectName { get; private set; }
+        public string OutputDirectory { get; private set; }
+
+        static List<string> Directories = new List<string>();
+
+        //Setup EventHandler
+        public void WireEventHandlers(Execute e)
+        {
+            ExecutionHandler handler = new ExecutionHandler(OnIncommingEventLog);
+            e.LogEvent += handler;
+        }
+
+        public void OnIncommingEventLog(object sender, EventLogger e)
+        {
+            CurrentLog = e.logs;
+        }
+
+        internal async Task TestTemplate(int i)
+        {
+            ProjectName = "TestProject";
+            OutputDirectory = temp;
+            Directories.Add("artifacts");
+            Directories.Add("build");
+            Directories.Add("docs");
+            Directories.Add("lib");
+            Directories.Add("samples");
+            Directories.Add("packages");
+            Directories.Add("test");
+
+            __.SelcectTemplate(i);
+            NETCommand = " new " + __.SelectedTemplate + " " + Options + "-o src/" + __.SelectedTemplate.Replace(" ", "_") + " -n " + __.SelectedTemplate.Replace(" ", "_");
+            Execute ExecuteClass = new Execute();
+            WireEventHandlers(ExecuteClass);
+            await ExecuteClass.CreateScript(OutputDirectory, Directories, NETCommand, ProjectName);
+        }
+    }
+    [Collection("Create All types of .NET Applications")]
+    public class FullTest
```

Comment:
Hatte es glaube schon mal erzählt Tests sind meistens in 
// Arrange

//Act

//Assert 

aufgeteilt für mehr Übersichtlichkeit bei dir geht das noch da du nur kurze Act's und Asserts hast, aber wenn die tests länger werden ist es cool wenn es eben etwas strukturierter ist.

Reply:
Bin immer noch der Meinung das es als Theory besser wäre

Reply:
@tomfrenzel  ping :P immer noch nicht hübsch

Reply:
!